### PR TITLE
[Dem] MoveAllMeshes python cleanup

### PR DESCRIPTION
**Pull request by ipouplana**
Summary:

- MoveAllMeshes is called at Predict method of sphere_strategy.py

- Some python warnings have been cleaned/updated

SwimmingDEMApp. and DemStructuresCouplingApp. tests are running ok.

### CODE REVIEW

**Comment by maceligueta** [comment]
If the python strategy starts to use Predict, I think that the DEM Analysis Stage should call it, too. (And the main_script.py, in case that it still exists or anyone still calls it)

**Comment by ipouplana** [comment]
The DEM_analysis_stage.py does not need anything else because the Predict is already called in the RunSolutionLoop of the analysis_stage.py. However, I have to add it in the MainScript.py

**Comment by maceligueta** [comment]
Thanks @ipouplana , I will check if the DEM tests run OK before approving.

**Comment by maceligueta** [comment]
We've got a problem with test_examples 32 and 33. Apparently they are failing in master branch.

**Comment by ipouplana** [comment]
But then, comparing the test results in the master branch with the test results in this branch, have you found any difference?

**Comment by farrufat-cimne** [comment]
->        if line.startswith("Particle_center_and_diameter"):
            data = ReadNextLine(f)
            center = [float(data[0]), float(data[1]), float(data[2])] 

the cluster being used does not fit the standard:

Name
benchmark_fibre
Begin centers_and_radii  
-1.0 0.0 0.0 0.5
-0.5 0.0 0.0 0.5
0.0 0.0 0.0 0.5
0.5 0.0 0.0 0.5
1.0 0.0 0.0 0.5
End centers_and_radii
     
Size
1.0

Volume
1.0

Inertia per unit mass
0.11875
0.6566
0.6566


This is how corncluster  standard cluster file looks like:

Name
corn3cluster3D

Begin centers_and_radii  
0.0 0.815  0.0 2.340
0.0 -0.870 0.0 1.872
0.0 -2.643 0.0 0.732
End centers_and_radii

Particle_center_and_diameter
0.000000 -0.110000 0.000000 6.530000

Size
6.530000

Volume
86.355672439

Inertia per unit mass
3.0074256
3.0074256
3.0074256

**Comment by maceligueta** [comment]
The problem with the tests will be solved in a different PR.

@ipouplana , can we add the Predict method to this file?
https://github.com/KratosMultiphysics/Kratos/blob/5a47ca58214d75e02c8b769c6487b66b25b8fecc/applications/DemStructuresCouplingApplication/python_scripts/dem_fem_coupling_explicit.py#L82
@KlausBSautter is maintaining this coupling file.

**Comment by KlausBSautter** [comment]
hi, I actually forgot about that file.... sry. This is handled in the CoSimApp now. So we might delete this file.

**Comment by maceligueta** [comment]
@KlausBSautter We have moved some functions to the Predict() in the DEM solver. Please make sure that you are calling that function, too.

**Comment by maceligueta** [comment]
@ipouplana  Let's delete this file and I will approve. Thanks for all the work you did here.

**Comment by KlausBSautter** [comment]
> @KlausBSautter We have moved some functions to the Predict() in the DEM solver. Please make sure that you are calling that function, too.

I am just using the DEM analysis stage in there.

**Comment by ipouplana** [comment]
If we delete dem_fem_coupling_explicit.py we could also delete a couple of deprecated methods (FinalizeSingleTimeStep, and AfterSolveOperations) that were only called in that file... Should I proceed ? Or do you want us to wait until the next PR?

**Comment by maceligueta** [comment]
> If we delete dem_fem_coupling_explicit.py we could also delete a couple of deprecated methods (FinalizeSingleTimeStep, and AfterSolveOperations) that were only called in that file... Should I proceed ? Or do you want us to wait until the next PR?

no rush. Approving.

**Review by maceligueta — DISMISSED** [review]

**Comment by maceligueta** [comment]
Oops, Travis is not passing tests!

**Comment by ipouplana** [comment]
Travis is failing... It seems that some tests fail...

File: applications/DEMApplication/python_scripts/DEM_procedures.py [hunk 1]
```diff
@@ -974,16 +974,16 @@ def evaluate_computation_of_fem_results():
...
+        message += '\nFunction \'MoveAllMeshes\' is deprecated. It is called inside sphere_strategy.py'
+        message += '\nIt will be removed after 10/31/2019.\n'
+        Logger.PrintWarning("DEM_procedures.py", message)
 
         spheres_model_part = all_model_parts.Get("SpheresPart")
         dem_inlet_model_part = all_model_parts.Get("DEMInletPart")
-        rigid_face_model_part = all_model_parts.Get("RigidFacePart")
-        cluster_model_part = all_model_parts.Get("ClusterPart")
 
-        self.mesh_motion.MoveAllMeshes(rigid_face_model_part, time, dt)
```
Comment (tteschemacher) [inline applications/DEMApplication/python_scripts/DEM_procedures.py:983]: just out of curiosity, why are you removing the rigid face model part from the movements?

Comment (maceligueta) [reply applications/DEMApplication/python_scripts/DEM_procedures.py:983]: It is done here:
https://github.com/KratosMultiphysics/Kratos/blob/6f91aeb60e3d5d665dacecc1e0c31ff232e209b9/applications/DEMApplication/custom_strategies/strategies/explicit_solver_strategy.cpp#L688
So we were doing it twice per time step (bug alert!).
We think that the strategy and scheme must take care of moving the walls. Depending on the combination of strategy and scheme, both the spheres and the walls should be moved in one step, or in two steps, or predicting and correcting iteratively, etc. Eventually, the MoveAllMeshes function should disappear, we are migrating it step by step.

Comment (maceligueta) [reply applications/DEMApplication/python_scripts/DEM_procedures.py:983]: I forgot to say that all walls (understanding a wall as a set of triangular or quadrilateral conditions) are converted into RigidBody elements at Initialize. If they are fixed, no motion is applied to them.

Comment (tteschemacher) [reply applications/DEMApplication/python_scripts/DEM_procedures.py:983]: Ok, thanks! 
I see your point. It seems that this is quite a complex topic within DEM.

**Comment by ipouplana** [comment]
> Oops, Travis is not passing tests!

I just merged master branch into this branch. Let's see if the error appears again...

**Comment by ipouplana** [comment]
The error appears again. Any clue? @maceligueta or @farrufat-cimne could it be related to this?

> We've got a problem with test_examples 32 and 33. Apparently they are failing in master branch.

Just in case, I copy the error here:

```
Running tests for DEMApplication

/home/travis/build/KratosMultiphysics/Kratos/applications//DEMApplication/tests/test_DEMApplication.py

.....EE...

======================================================================

ERROR: test_Analytics_1 (test_analytics.TestAnalytics)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/KratosMultiphysics/Kratos/applications/DEMApplication/tests/test_analytics.py", line 181, in test_Analytics_1

    CreateAndRunStageInOneOpenMPThread(AnalyticsTestSolution, model, parameters_file_name)

  File "/home/travis/build/KratosMultiphysics/Kratos/applications/DEMApplication/tests/test_analytics.py", line 25, in CreateAndRunStageInOneOpenMPThread

    my_obj(model, project_parameters).Run()

  File "/home/travis/build/KratosMultiphysics/Kratos/kratos/python_scripts/analysis_stage.py", line 50, in Run

    self.RunSolutionLoop()

  File "/home/travis/build/KratosMultiphysics/Kratos/kratos/python_scripts/analysis_stage.py", line 70, in RunSolutionLoop

    self.OutputSolutionStep()

  File "/home/travis/build/KratosMultiphysics/Kratos/applications/DEMApplication/python_scripts/DEM_analysis_stage.py", line 538, in OutputSolutionStep

    self.FinalizeTimeStep(self.time)

  File "/home/travis/build/KratosMultiphysics/Kratos/applications/DEMApplication/tests/test_analytics.py", line 62, in FinalizeTimeStep

    self.CheckValueOfNormalImpactVelocity(normal_impact_vel, expected_value, tolerance)

  File "/home/travis/build/KratosMultiphysics/Kratos/applications/DEMApplication/tests/test_analytics.py", line 89, in CheckValueOfNormalImpactVelocity

    raise ValueError('Incorrect value for NORMAL_IMPACT_VELOCITY: expected value was '+ str(expected_value) + ' but received ' + str(normal_impact_vel))

ValueError: Incorrect value for NORMAL_IMPACT_VELOCITY: expected value was 14.4637 but received 3.0

======================================================================

ERROR: test_Glued_Particles_1 (test_glued_particles.TestGluedParticles)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/KratosMultiphysics/Kratos/applications/DEMApplication/tests/test_glued_particles.py", line 78, in test_Glued_Particles_1

    CreateAndRunStageInOneOpenMPThread(GluedParticlesTestSolution, model, parameters_file_name)

  File "/home/travis/build/KratosMultiphysics/Kratos/applications/DEMApplication/tests/test_glued_particles.py", line 25, in CreateAndRunStageInOneOpenMPThread

    my_obj(model, project_parameters).Run()

  File "/home/travis/build/KratosMultiphysics/Kratos/kratos/python_scripts/analysis_stage.py", line 50, in Run

    self.RunSolutionLoop()

  File "/home/travis/build/KratosMultiphysics/Kratos/kratos/python_scripts/analysis_stage.py", line 70, in RunSolutionLoop

    self.OutputSolutionStep()

  File "/home/travis/build/KratosMultiphysics/Kratos/applications/DEMApplication/python_scripts/DEM_analysis_stage.py", line 538, in OutputSolutionStep

    self.FinalizeTimeStep(self.time)

  File "/home/travis/build/KratosMultiphysics/Kratos/applications/DEMApplication/tests/test_glued_particles.py", line 46, in FinalizeTimeStep

    self.CheckValue("Angular Velocity at time "+ str(time), angular_velocity[0], 2.0, tolerance)

  File "/home/travis/build/KratosMultiphysics/Kratos/applications/DEMApplication/tests/test_glued_particles.py", line 61, in CheckValue

    raise ValueError('Incorrect value for ' + explaining_string +': expected value was '+ str(expected_value) + ' but received ' + str(value))

ValueError: Incorrect value for Angular Velocity at time 0.010000000000000002: expected value was 2.0 but received 0.0

----------------------------------------------------------------------

Ran 10 tests in 29.947s

FAILED (errors=2)
```
There is also an error in the StructuralMechanics tests:

```
Running tests for StructuralMechanicsApplication

/home/travis/build/KratosMultiphysics/Kratos/applications//StructuralMechanicsApplication/tests/test_StructuralMechanicsApplication.py

.

----------------------------------------------------------------------

Ran 1 test in 2.297s

OK

E

======================================================================

ERROR: test_execution (structural_mechanics_test_factory_mpi.ShellT3AndQ4LinearStaticStructPinchedCylinderTests)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/KratosMultiphysics/Kratos/applications/StructuralMechanicsApplication/tests/structural_mechanics_test_factory.py", line 55, in test_execution

    self.test.RunSolutionLoop()

  File "/home/travis/build/KratosMultiphysics/Kratos/kratos/python_scripts/analysis_stage.py", line 69, in RunSolutionLoop

    self.FinalizeSolutionStep()

  File "/home/travis/build/KratosMultiphysics/Kratos/kratos/python_scripts/analysis_stage.py", line 144, in FinalizeSolutionStep

    process.ExecuteFinalizeSolutionStep()

  File "/home/travis/build/KratosMultiphysics/Kratos/kratos/python_scripts/from_json_check_result_process.py", line 132, in ExecuteFinalizeSolutionStep

    values_json = self.data[node_identifier][variable_name]

KeyError: 'NODE_1'

----------------------------------------------------------------------

Ran 1 test in 2.298s

FAILED (errors=1)
```
although I don't know what relation it may have with this PR...

**Comment by farrufat-cimne** [comment]
the walls in test are not moving.

**Comment by ipouplana** [comment]
> the walls in test are not moving.

It seems we can't erase this from MoveAllMeshes:

`self.mesh_motion.MoveAllMeshes(rigid_face_model_part, time, dt)`

File: applications/DEMApplication/python_scripts/DEM_procedures.py (hunk 1 above)
Comment (maceligueta) [reply applications/DEMApplication/python_scripts/DEM_procedures.py:983]: @tteschemacher  It seems like I was totally wrong. That line only moves the walls if they form a RigidBody which can move freely, or with some restricted dofs. If the walls are totally fixed then they must be moved by MoveAllMeshes.
When you see this strange mixes we are doing, you realize how importat migrating to processes is. However, they have a problem: processes can't get into the Strategy or the scheme, they can only be called by the AnalysisStage at InitializeSolutionStep, FinalizeSolutionStep and so on...

**Review by maceligueta — APPROVED** [review]

**Pull request merged**